aidial_adapter_bedrock/llm/message.py

```diff
@@ -1,95 +1,260 @@
-from typing import List, Optional, Union
-
-from aidial_sdk.chat_completion import (
-    CustomContent,
-    FunctionCall,
-    Message,
-    Role,
-    ToolCall,
-)
+from abc import ABC, abstractmethod
+from typing import List, Optional, Self, Union
+
+from aidial_sdk.chat_completion import Attachment, CustomContent, FunctionCall
+from aidial_sdk.chat_completion import Message as DialMessage
+from aidial_sdk.chat_completion import MessageContentPart, Role, ToolCall
 from pydantic import BaseModel
 
+from aidial_adapter_bedrock.dial_api.request import (
+    collect_text_content,
+    is_plain_text_content,
+    is_text_content_parts,
+)
 from aidial_adapter_bedrock.llm.errors import ValidationError
 
 
-class SystemMessage(BaseModel):
-    content: str
+class MessageABC(ABC, BaseModel):
+    @abstractmethod
+    def to_message(self) -> DialMessage: ...
 
-    def to_message(self) -> Message:
-        return Message(role=Role.SYSTEM, content=self.content)
+    @classmethod
+    @abstractmethod
+    def from_message(cls, message: DialMessage) -> Self | None: ...
 
 
-class HumanRegularMessage(BaseModel):
-    content: str
+class BaseMessageABC(MessageABC):
+    @property
+    @abstractmethod
+    def text_content(self) -> str: ...
+
+
+class SystemMessage(BaseMessageABC):
+    content: str | List[MessageContentPart]
+
+    def to_message(self) -> DialMessage:
+        return DialMessage(
+            role=Role.SYSTEM,
+            content=self.content,
+        )
+
+    @classmethod
+    def from_message(cls, message: DialMessage) -> Self | None:
+        if message.role != Role.SYSTEM:
+            return None
+
+        content = message.content
+
+        if content is None:
+            raise ValidationError("System message is expected to have content")
+
+        if isinstance(content, str):
+            return cls(content=content)
+
+        if is_text_content_parts(content):
+            return cls(content=content)  # type: ignore
```

Comment:
If we add this typeguard function here it would be intuitive to specify that we don't allow anything except text content part in system message
```python
class SystemMessage(...):
    content: str | List[MessageContentTextPart]
```

Reply:
Fixed